src/bulls_and_cows_lib/board.cpp

```diff
@@ -4,5 +4,189 @@
 namespace bulls_and_cows {
 
     // TODO: define the body of the functions declared in board.cpp
+    // Create a board and with a new random secret code composed of allowed characters
+    Board create_board(const GameOptions& game_options)
+    {
+        Board board;
+         
+
+        for (int unsigned i = 0; i < game_options.number_of_characters_per_code; i++)
+        {
+            board.secret_code.value =
+                board.secret_code.value + generate_random_character(game_options.minimum_allowed_character,
+                                                                    game_options.maximum_allowed_character);
+        }
+
+        return board;
+    }
+
+    // Validate that a user-entered code contains proper number of allowed characters
+    bool validate_attempt(const GameOptions& game_options, const Code& attempt)
+    {
+      
+
+        if (attempt.value.size() == game_options.number_of_characters_per_code)
+        {
+            for (int unsigned i = 0; i < game_options.number_of_characters_per_code; i++)
+            {
+                if (attempt.value[i] < game_options.minimum_allowed_character ||
+                    attempt.value[i] > game_options.maximum_allowed_character)
+                {
+                    return false;
+                }
+            }
+                return true;
+        }
+        return false;
+        
+    }
+
+ 
+
+    // Compare a user-entered code with the secret code and give the corresponding number of bulls and cows
+    Feedback compare_attempt_with_secret_code(Code attempt,Code secret_code)
+    {
+       
+        Feedback feedback;
+       
+
+        for (int unsigned i = 0; i < attempt.value.size();i++)
+        {
+            if (attempt.value[i] == secret_code.value[i])
+            {
+
+                attempt.value.erase(i, 1);
+                secret_code.value.erase(i, 1);
+
+                feedback.bulls++;
+
+                i--;
+            }
+           
+        }
+
+        for (int unsigned i = 0; i < secret_code.value.size(); i++)
+        {
```

Comment:
La double boucle me semble très suspecte.

Tu n'as pas mis d'accolade pour le corps du deuxième for, c'est possible, mais ça rend le code moins lisible dans ce cas, car en +, tu as mis le if au même niveau d'indentation que le for.

Et surtout :
La variable i itère sur les indices du secret_code, et la variable j itère sur les indices de l'attempt.
Mais tu fais i-- lorsque tu supprimes un élément de l'attempt.

Et pourquoi n'as tu pas conservé gardé une logique qui utilisait la fonction membre find ?

Reply:
Done
Je n'ai pas réussi à déboguer avec find donc j'ai créer les deux boucles for pour avancer